File: spina_portal/tests/client-renewal-authority-wording.test.mjs
import assert from 'node:assert/strict';
import { readFile } from 'node:fs/promises';
import test from 'node:test';

const clientRoleSource = await readFile(
  new URL('../assets/roles/client.js', import.meta.url),
  'utf8',
);

test('Client Web renewal wording keeps assigned Collector recommendation before Management decision', () => {
  assert.match(
    clientRoleSource,
    /permanently assigned Collector must recommend/i,
  );
  assert.match(
    clientRoleSource,
    /before Management reviews and decides/i,
  );
});

test('Client Web renewal wording keeps every other required signer on their own SPINA account', () => {
  assert.match(
    clientRoleSource,
    /other required signer must use their own SPINA account/i,
  );
});

File: gilbic_mobile/test/client_renewal_authority_wording_test.dart
import 'dart:io';

import 'package:flutter_test/flutter_test.dart';

void main() {
  final renewalModelSource = File(
    'lib/src/core/renewals/renewal_request.dart',
  ).readAsStringSync();
  final renewalPageSource = File(
    'lib/src/features/client/client_renewal_page.dart',
  ).readAsStringSync();
  final renewalWorkflowSource = File(
    'lib/src/features/client/client_renewal_workflow_page.dart',
  ).readAsStringSync();

  test('Client Android renewal wording preserves Collector then Management sequence', () {
    expect(
      renewalModelSource,
      contains("'Pending Collector recommendation / Management review'"),
    );
    expect(
      renewalPageSource,
      contains(
        'Your assigned Collector must recommend it before Management review.',
      ),
    );
  });

  test('Client Android renewal wording keeps other required signers on their own account', () {
    const signerCopy =
        'Any other required signer must use their own SPINA account.';
    expect(renewalPageSource, contains(signerCopy));
    expect(renewalWorkflowSource, contains(signerCopy));
  });
}
